### PR TITLE
Merge custom permissions for Glance and Cinder pools, (bnc#944810)

### PR DESCRIPTION
**Pull request by mjura**

### CODE REVIEW

**Comment by mjura** [comment]
Was tested and works fine, but I wouldn't like to merge this change

**Comment by mjura** [comment]
Change https://github.com/crowbar/crowbar-openstack/pull/48 should be  sufficient for fixing bnc#944810

File: chef/cookbooks/ceph/providers/client.rb [hunk 1]
```diff
@@ -68,7 +68,24 @@ def auth_set_key(ceph_conf, admin_keyring, keyname, caps)
     Chef::Log.info("Deleting old key with incorrect caps")
     # don't remove custom existing caps for osd pools
     caps_osd = get_caps(ceph_conf, admin_keyring, keyname)["osd"] + "," + caps["osd"]
-    caps["osd"] = caps_osd.split(",").collect(&:strip).uniq.join(",")
+    cap_osd = []
+    cap_hash = {}
+    caps_osd.split(",").collect(&:strip).uniq.each do |cap|
+      if cap =~ /pool/
```
Comment (vuntz) [inline chef/cookbooks/ceph/providers/client.rb]: Should probably be `/ pool=/`

File: chef/cookbooks/ceph/providers/client.rb [hunk 2]
```diff
@@ -68,7 +68,24 @@ def auth_set_key(ceph_conf, admin_keyring, keyname, caps)
...
+        cap_tmp = cap.split(" ")
+        if cap_hash[cap_tmp[2]]
+          cap_hash[cap_tmp[2]] = (cap_hash[cap_tmp[2]] + cap_tmp[1]).chars.to_a.sort.uniq.join
+        else
+          cap_hash[cap_tmp[2]] = cap_tmp[1]
```
Comment (vuntz) [inline chef/cookbooks/ceph/providers/client.rb]: You're assuming that it's always `allow`. Is that true?

File: chef/cookbooks/ceph/providers/client.rb [hunk 3]
```diff
@@ -68,7 +68,24 @@ def auth_set_key(ceph_conf, admin_keyring, keyname, caps)
     Chef::Log.info("Deleting old key with incorrect caps")
     # don't remove custom existing caps for osd pools
     caps_osd = get_caps(ceph_conf, admin_keyring, keyname)["osd"] + "," + caps["osd"]
-    caps["osd"] = caps_osd.split(",").collect(&:strip).uniq.join(",")
+    cap_osd = []
+    cap_hash = {}
+    caps_osd.split(",").collect(&:strip).uniq.each do |cap|
+      if cap =~ /pool/
+        cap_tmp = cap.split(" ")
```
Comment (vuntz) [inline chef/cookbooks/ceph/providers/client.rb:75]: I'd probably add some variables for readability:

```
pool = cap_tmp[2]
cap = cap_tmp[1]
```

File: chef/cookbooks/ceph/providers/client.rb (hunk 2 above)
Comment (mjura) [inline chef/cookbooks/ceph/providers/client.rb]: Yes, format of capabilities is known and always this same.

**Comment by toabctl** [comment]
> Was tested and works fine, but I wouldn't like to merge this change

Is this still true, @mjura  ? Or do you want to have this merged? If not, please close the PR or add a "do not merge yet" label

**Comment by mjura** [comment]
@toabctl : Yes, we still waiting for feedback from customer. I still don't want to merge it.

**Comment by mjura** [comment]
Bug was closed, this change is not needed any more

**Pull request closed without merging**